recipes/compute_library/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+            "clang": "5",
+            "apple-clang": "5.1",
+        }
+
+    def config_options(self):
+        # INFO: Neon option is reserved to arm architecture
+        if "arm" not in str(self.settings.arch):
+            del self.options.enable_neon
+        # INFO: OpenMP option only works with g++, according to the documentation
+        if self.settings.compiler == "clang":
+            del self.options.enable_openmp
+        # INFO: OpenCL fails to build with MacOS
+        if self.settings.os == "Macos":
+            self.options.enable_opencl = False
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        self.tool_requires("scons/4.3.0")
+
+    def requirements(self):
+        if self.options.enable_opencl:
+            self.requires("opencl-headers/2023.04.17")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.")
+        supported_os = ["Android", "Linux", "OpenBSD", "Macos", "Tizen"]
+        if str(self.settings.os) not in supported_os:
+            raise ConanInvalidConfiguration(f"{self.ref} does not support {self.settings.os}. It is only supported on {supported_os}.")
+        if "arm" not in str(self.settings.arch) and "x86" not in str(self.settings.arch):
+            raise ConanInvalidConfiguration(f"{self.ref} does not support {self.settings.arch}. It is only supported on arm and x86.")
+        if "x86" in str(self.settings.arch) and (self.options.get_safe("enable_neon") or not self.options.enable_opencl):
+            raise ConanInvalidConfiguration(f"{self.ref} can be built for x86_64 targets only with enable_neon=False and enable_opencl=True.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = AutotoolsDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    def build(self):
+        # INFO: Using scons to build the library we don't have control over shared/static and install step, it is done all together always
+        # INFO: https://arm-software.github.io/ComputeLibrary/latest/how_to_build.xhtml
+        yes_no = lambda v: "1" if v else "0"
+        debug = yes_no(self.settings.build_type == "Debug")
+        build_os = str(self.settings.os).lower()
+        arch = {"armv8": "armv8a", "x86": "x86_32", "armv7": "armv7a", "armv7hf": "armv7a-hf"}.get(str(self.settings.arch), str(self.settings.arch))
```

Comment:
ARM compute allows to build code with armv8.2a (fp16 support), SVE and SME.
How to use it via this recipe?

Reply:
for example Apple M1, M2 support fp16, so we need to enable it by default.